sdk/core/azure-core/tests/testserver_tests/async_tests/test_rest_headers_async.py

```diff
@@ -7,9 +7,124 @@
 import pytest
 from azure.core.rest import HttpRequest
 
+# flask returns these response headers, which we don't really need for these following tests
+RESPONSE_HEADERS_TO_IGNORE = [
+    "Content-Type",
+    "Content-Length",
+    "Server",
+    "Date",
+]
+
+@pytest.fixture
+def get_response_headers(client):
+    async def _get_response_headers(request):
+        response = await client.send_request(request)
+        response.raise_for_status
+        for header in RESPONSE_HEADERS_TO_IGNORE:
+            response.headers.pop(header)
+        return response.headers
+    return _get_response_headers
+
+@pytest.mark.asyncio
+async def test_headers_response(get_response_headers):
+    h = await get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    assert "a" in h
+    assert "A" in h
+    assert "b" in h
+    assert "B" in h
+    assert "c" not in h
+    assert h["a"] == "123, 456"
+    assert h.get("a") == "123, 456"
+    assert h.get("nope", default=None) is None
```

Comment:
Could we add a test for this that doesn't set `default`?

Reply:
yep!